specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkSecurityPerimeter.json

```diff
@@ -1369,6 +1406,24 @@
       },
       "description": "Properties of network security perimeter."
     },
+    "UpdateTagsRequest": {
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

Resource id can't be updated by patch.

Reply:
This field is a readonly field. We use content in the URL to set this field in the controller and pass it to the operations layer. It is not being used to update the Id. Similar to put nsp profile

Reply:
Even if the client sets this field, it will be reset with the content in the URL, passed to the operation layer to do a get on the resource and update the tags. Internally, we only use it to perform a get.

Reply:
Basically this is not required, as you can access the URL/referrer anyways.

Reply:
thats correct but the object is needed in the operations layer to access the NSP with the specified id